### DISCUSSION TITLE
Unity SDK Bugs

### DISCUSSION THREAD

**Discussion by katpavan** [Web3Auth/Web3Auth#1125] [Help]
### We're using the unity sdk

- We get this screen sometimes when logging in the mobile unity based game. We recently switched to web3auth production env (aka Cyan) and haven't seen this yet. But we're testing.

<img width="258" alt="image" src="https://user-images.githubusercontent.com/45500874/209270435-7b6f9048-ad3f-4fe9-af9c-19c3af137717.png">

- apple login failing for whatever reason - shows us a web3auth error page
- Facebook login failed. api error msg: 

`Argument email for data.email is missing.\n","stack":"Error: Argument email for data.email is missing.` 

This means facebook isn’t returning the email.

- one of our engineers got this when trying to login with google, and when he clicks the button "redirect to app" it goes nowhere.
<img width="212" alt="image" src="https://user-images.githubusercontent.com/45500874/209270774-b39e6890-dadf-450e-a735-1e0509946a4d.png">

- doesnt use the saved google accounts i have on my mobile device but other games and apps on my phone bring up my saved google accounts to make it easy to sign in:

<img width="230" alt="image" src="https://user-images.githubusercontent.com/45500874/209270849-08e9e448-3a81-42e3-8ae4-e130031540ce.png">

- we would love introductions to other companies using the unity SDK so we can share notes and integration tips
- we're looking forward to refresh tokens. Would this be available early January as well for the unity SDK?
- how do we add our logo into the web3auth login screen?

**Comment by dangdennis** [comment]
Regarding facebook oauth, can we change perms via the sdk to request for their email as well without setting up our own custom auth?

**Comment by jasonl987** [comment]
![IMG_5562](https://user-images.githubusercontent.com/5432857/209294519-cd1e2ae1-9d44-4d2e-aa8b-4f5008b6ee55.png)

also received this error while trying to auth using apple ID

also fwiw sometimes when I hard restart the app, then try loggin in web3auth will then work. not sure why that is
